test/Guardian.t.sol

```diff
@@ -0,0 +1,367 @@
+        stdstore.target(address(lendingPool)).sig(lendingPool.pauseTimestamp.selector).checked_write(uint256(0));
+        // Warp the block timestamp to 60days for smooth testing
+        vm.warp(60 days);
+    }
+
+    function testSuccess_changeGuardian() public {
+        // Given: the lending pool owner is owner§
+        vm.startPrank(owner);
+        // When: the owner changes the guardian
+        lendingPool.changeGuardian(newGuardian);
+        vm.stopPrank();
+        // Then: the guardian is changed
+        assertEq(lendingPool.guardian(), newGuardian);
+
+        // When: The owner changes the guardian back to the original guardian
+        vm.startPrank(owner);
+        lendingPool.changeGuardian(guardian);
+        vm.stopPrank();
+
+        // Then: the guardian is changed
+        assertEq(lendingPool.guardian(), guardian);
+    }
+
+    function testRevert_changeGuardian_onlyOwner() public {
+        // Given: the lending pool owner is owner
+        vm.startPrank(nonOwner);
+        // When: a non-owner tries to change the guardian, it is reverted
+        vm.expectRevert("UNAUTHORIZED");
+        lendingPool.changeGuardian(guardian);
+        vm.stopPrank();
+        // Then: the guardian is not changed
+        assertEq(lendingPool.guardian(), guardian);
+    }
+
+    function testRevert_pause_onlyGuard() public {
+        // Given When Then: the lending pool is paused
+        vm.expectRevert("Guardian: Only guardian can call this function");
+        vm.startPrank(owner);
+        lendingPool.pause();
+        vm.stopPrank();
+    }
+
+    function testSuccess_depositGuarded_notPause() public {
+        // Given: the lending pool is not paused
+        vm.startPrank(user);
+        // When: a user supplies
+        lendingPool.depositGuarded(100);
+        vm.stopPrank();
+        // Then: the total supply is updated
+        assertEq(lendingPool.totalSupply(), 100);
+    }
+
+    function testRevert_depositGuarded_paused() public {
+        // Given: the lending pool supply is paused, only supply paused
+        vm.startPrank(guardian);
+        lendingPool.pause();
+        vm.stopPrank();
+
+        // When Then: a user tries to supply, it is reverted as paused
+        vm.expectRevert("Guardian: deposit paused");
+        vm.startPrank(user);
+        lendingPool.depositGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total supply is not updated
+        assertEq(lendingPool.totalSupply(), 0);
+
+        // When: owner can unPauses the borrow
+        vm.startPrank(owner);
+        lendingPool.unPause(true, true, false, true, true);
+        vm.stopPrank();
+
+        // Then: user tries to borrow, which is not paused
+        vm.startPrank(user);
+        lendingPool.borrowGuarded(100);
+        vm.stopPrank();
+
+        // Then: the total borrow is updated
+        assertEq(lendingPool.totalBorrow(), 100);
+
+        // Revert: the lending pool is unpaused
```

Comment:
is this comment wrong or should this revert?

Reply:
It is not same revert as expectRevert to remove the effects of the pause actions in the test. I move it to setup with stdstore now